public/js/deleteCourseModal.js

```diff
@@ -0,0 +1,59 @@
+let deleteButtons = document.querySelectorAll('.btnDeleteCourse');
+for (let db of deleteButtons) {
+    db.addEventListener("click", () => {
+        const courseId = db.getAttribute('data-courseId');
+        displayDetails('./api/deleteCourse/' + courseId, courseId);
+    });
+}
+
+function displayDetails(url, courseId) {
+    fetch(url).then(
+        response => {
+            return response.json();
+        }).then(data => {
+            let studentNamesContainer = document.querySelector('#studentNamesContainer');
+            studentNamesContainer.innerHTML = '<h4>Name</h4>';
+            let DeleteCoursesForm = document.querySelector('#DeleteCoursesForm');
+            DeleteCoursesForm.innerHTML = '';
+            DeleteCoursesForm.action = "api/deleteCourse/" + courseId;
+            let prettyCourses = makeCoursesPretty(data['availableCourses'])
+            for (let student of data['applicants']) {
+                generateStudents(student['name'], studentNamesContainer);
+                generateOptions(prettyCourses, DeleteCoursesForm, student['id'], data['availableCourses']);
+            }
+        }
+    )
+}
+
+function generateStudents(name, studentNamesContainer) {
+    let studentName = document.createElement('h5');
+    studentName.innerText = name;
+    studentNamesContainer.appendChild(studentName);
+}
+
+function makeCoursesPretty(courses) {
```

Comment:
This name doesn't really describe what the function does. Maybe something like `formatCourseNameAndStart` is less exciting but more descriptive!